simulation/behavior_tree_plugin/include/behavior_tree_plugin/transition_events/transition_event.hpp

```diff
@@ -36,7 +36,7 @@ class TransitionEvent
   BT::TimePoint first_timestamp_;
   std::vector<BT::TreeNode::StatusChangeSubscriber> subscribers_;
   BT::TimestampType type_;
-  std::string current_action_;
+  std::string current_action_{"waiting"};
```

Comment:
I've added this change to provide that the status `std::string` is never empty and its content is consistent with that is returned by the `EntityManager` when `npc_logic_started_` is false (unstarted).

Since the `EntityManager` returns a `“waiting”` status bypassing the `BehaviorTree`, I believe that then the `BehaviorTree` should also have the same status. What do you think about this? :bow: 

https://github.com/tier4/scenario_simulator_v2/blob/081643a1213c83ad9da54f47c136121a6a64f340/simulation/traffic_simulator/src/entity/entity_manager.cpp#L325-L336

Reply:
This `waiting` is waiting for Autoware to initialize.
Entities other than EgoEntity do not have an Autoware initialization step, so the `waiting` state is inappropriate.

Reply:
If my understanding is correct, the result of this implementation:

- https://github.com/tier4/scenario_simulator_v2/blob/081643a1213c83ad9da54f47c136121a6a64f340/simulation/traffic_simulator/src/entity/entity_manager.cpp#L328-L330

- is that the `EntityManager::getCurrentAction()` method returns a hard-coded `“waiting”` for any `non-Ego Entity` - when `npc_logic_started_ == False`. 

- State when `npc_logic_started_ == False` is the period when Autoware is in the initialization state, then the state of `non-Ego Entities` is equal to `“waiting”`.

- It follows that it is expected that the **default** `current_action` for `non-Ego Entities` is just equal to `“waiting”`.

It is possible that `EntityManager::getCurrentAction()` should return `“none”` for the mentioned cases instead of `“waiting”`- if this is your intention let me know please. :bow: 

But, for now I'll set `“none”` in `TransitionEvent::current_action_` as you suggested, since it's possible that it's indeed not a good place to set the **default** state of `BehaviorTree` for any Entity (the state before executing any transition).